src/app/chat/views/messages.nim

```diff
@@ -480,3 +480,16 @@ QtObject:
       self.searchedMessageLoaded(messageId)
     else:
       self.loadMessagesUntillMessageWithIdIsLoaded(messageId)
+
+  proc downloadMessages*(self: MessageView, filePath: string) {.slot.} =
+    let messages = newJArray()
+    for message in self.messageList[self.channelView.activeChannel.id].messages:
+      if message.id == "":
+        continue
+
+      messages.elems.add(%*{
+        "id": message.id, "text": message.text, "clock": message.clock,
+        "alias": message.alias, "from": message.fromAuthor
+      })
+
+    writeFile(replace(filePath, "file://", ""), $messages)
```

Comment:
Please don't do any magic with `file://`, please please use QUrl::toLocalFile and/or QUrl::fromLocalFile if you need a conversion between QUrl and native disk filepath.

Reply:
There is another pending PR here https://github.com/status-im/status-desktop/pull/3534#issuecomment-919432339 having the same issue. Maybe it is time to expose these two conversion functions from DOtherSide?

Reply:
@pinebit updated pls let me know if that's better for you!

Reply:
Perfect! Thanks!